lighthouse-core/test/network-records-to-devtools-log.js

```diff
@@ -115,7 +115,8 @@ function getResponseReceivedEvent(networkRecord, index) {
         connectionId: networkRecord.connectionId || 140,
         fromDiskCache: networkRecord.fromDiskCache || false,
         fromServiceWorker: networkRecord.fetchedViaServiceWorker || false,
-        encodedDataLength: networkRecord.transferSize || 0,
+        encodedDataLength: typeof networkRecord.transferSize === 'undefined' ?
```

Comment:
I didn't understand the note about `undefined`. Why does this need to change?

Don't need to do typeof here, just `networkRecord.transferSize === undefined` works. Or if the data type is somehow uncertain, flip the logic to `typeof networkRecord.transferSize === 'number' ? networkRecord.transferSize : 0`.

Reply:
> Don't need to do typeof here

ah yes, old habits :)

> Or if the data type is somehow uncertain, flip the logic to

The spirit of the original I assumed was `?? 0`, i.e. use `0` as a default if some value is not explicitly set. That seems better than ignoring all values that aren't numbers.

> I didn't understand the note about undefined. Why does this need to change?

If we are ever to test unexpected values coming from the protocol, we need a mechanism to mock those values in the devtools log. The affected code was attempting to handle such unexpected values.